routes/api.js

```diff
@@ -199,8 +216,10 @@ var killKernel = function(kernelId) {
         method: 'DELETE',
         headers: headers
     }, function(err, response, body) {
-        debug('PROXY: kill kernel response: ' +
-            response.statusCode + ' ' + response.statusMessage);
+        if (response) {
```

Comment:
What happens if we don't have a response here? Is that an error? If so, probably should be logged at the very least. If there is an error, is there anything we should do to recover/retry?